### PR TITLE
Syntax Update

### PR DESCRIPTION
**Pull request by SgtGremulock**
Got bored so I decided to update the syntax for this. Go nuts.

### CODE REVIEW

File: game/sourcemod/scripting/donationcontrol.sp [hunk 1]
```diff
@@ -4,187 +4,183 @@
...
+	for (int i = 0; i < sizeof(hConVars); i++)
 	{
-		HookConVarChange(hConVars[i], HandleCvars);
+		hConVars[i].AddChangeHook(HandleCvars);
 	}
 
 	RegConsoleCmd("sm_donate", DonatePanel);
 
-	//AutoExecConfig(true, "plugin.donation.control");
+	AutoExecConfig();
```
Comment (RumbleFrog) [inline game/sourcemod/scripting/donationcontrol.sp]: `autoCreate` and `name` parameter should be passed here?

Also, any idea why was it commented out previously?

File: game/sourcemod/scripting/donationcontrol.sp [hunk 2]
```diff
@@ -4,187 +4,183 @@
...
+public void HandleCvars(ConVar cvar, const char[] sOldValue, const char[] sNewValue)
 {
-	if (StrEqual(sOldValue, sNewValue, true)) return;
+	if (StrEqual(sOldValue, sNewValue, true))
+		return;
 
-	new iNewValue = StringToInt(sNewValue);
+	int iNewValue = StringToInt(sNewValue);
 
 	if (cvar == hConVars[0])
```
Comment (RumbleFrog) [inline game/sourcemod/scripting/donationcontrol.sp]: While here, bonus points if you convert these to switch cases

**Comment by RumbleFrog** [comment]
Looks good, left some change requests, if you could look them over.

File: game/sourcemod/scripting/donationcontrol.sp (hunk 1 above)
Comment (SgtGremulock) [reply game/sourcemod/scripting/donationcontrol.sp]: The parameters by default for AutoExecConfig pass true for the first parameter, and the name of the config file defaults to "plugin.donationcontrol.cfg". Therefore, setting those 2 parameters is unnecessary.

File: game/sourcemod/scripting/donationcontrol.sp (hunk 2 above)
Comment (SgtGremulock) [reply game/sourcemod/scripting/donationcontrol.sp]: You can't do switch cases with handles.

Comment (RumbleFrog) [reply game/sourcemod/scripting/donationcontrol.sp]: Thought you could coerce it to int primitive, but I guess not.

**Pull request merged**